eng/breakingChanges/breaking-change-doc.ps1

```diff
@@ -0,0 +1,1114 @@
+
+        if ($response.Count -eq 0) {
+            Write-Error "❌ No example breaking change issues found in $($Config.DocsRepo) with label 'breaking-change'"
+            Write-Error "   Examples are required for high-quality documentation generation. Please check repository and label."
+            exit 1
+        }
+
+        $examples = @()
+        foreach ($issue in $response) {
+            $examples += @"
+**Example #$($issue.number)**: $($issue.title)
+URL: $($issue.html_url)
+Body: $(Limit-Text -text $issue.body -maxLength 800)
+"@
+        }
+        return $examples -join "`n`n---`n`n"
+    }
+    catch {
+        Write-Error "❌ Failed to fetch example breaking change issues from $($Config.DocsRepo): $($_.Exception.Message)"
+        Write-Error "   Examples are required for high-quality documentation generation. Please check repository access."
+        exit 1
+    }
+}
+
+# Function to find the closest tag by commit distance
+function Find-ClosestTagByDistance {
+    param([string]$targetCommit, [int]$maxTags = 10)
+
+    $recentTags = git tag --sort=-version:refname 2>$null | Select-Object -First $maxTags
+    $closestTag = $null
+    $minDistance = [int]::MaxValue
+
+    foreach ($tag in $recentTags) {
+        # Check if this tag contains the target commit (skip if it does for merged PRs)
+        if ($targetCommit -match '^[a-f0-9]{40}$') {
+            # This is a commit hash, check if tag contains it
+            git merge-base --is-ancestor $targetCommit $tag 2>$null
+            if ($LASTEXITCODE -eq 0) {
+                # This tag contains our commit, skip it
+                continue
+            }
+        }
+
+        # Calculate commit distance between tag and target
+        $distance = git rev-list --count "$tag..$targetCommit" 2>$null
+        if ($LASTEXITCODE -eq 0 -and $distance -match '^\d+$') {
+            $distanceNum = [int]$distance
+            if ($distanceNum -lt $minDistance) {
+                $minDistance = $distanceNum
+                $closestTag = $tag
+            }
+        }
+    }
+
+    return $closestTag
+}
+
+# Function to get version information using local git repository
+function Get-VersionInfo {
+    param([string]$prNumber, [string]$mergedAt, [string]$baseRef = "main")
+
+    try {
+        # Change to the repository directory (we're already in the repo)
+        Push-Location $repoRoot
+
+        try {
+            # Ensure we have latest info
+            git fetch --tags 2>$null | Out-Null
+
+            # For merged PRs, try to find the merge commit and get version info
+            if ($prNumber -and $mergedAt) {
+                # Get the merge commit for this PR
+                $mergeCommit = gh pr view $prNumber --repo $Config.SourceRepo --json mergeCommit --jq '.mergeCommit.oid' 2>$null
+
+                if ($mergeCommit) {
```

Comment:
By limiting this to only run on merged PRs, we can drop a lot of this logic too and not worry about scenarios of walking arbitrary git histories.